src/chart/speclayout/tablechart.h

```diff
@@ -18,14 +18,25 @@ template <typename Item>
 void TableChart::setupVector(std::vector<Item> &items,
     bool singleColumn)
 {
-	if (items.empty()) return;
-
 	auto size = 0;
 	for (auto &item : items)
 		if (item.enabled) ++size;
 
-	auto rowsize =
-	    singleColumn ? 1 : static_cast<ssize_t>(ceil(sqrt(size)));
+	if (size == 0) return;
+
+	ssize_t rowsize{1};
+	if (!singleColumn) {
+		rowsize = static_cast<ssize_t>(ceil(sqrt(size)));
+		auto sq = static_cast<ssize_t>(ceil(sqrt(
+		    16.0 * static_cast<double>(rowsize * rowsize) / 9.0)));
```

Comment:
we should use 1:1 ratio to 
- prevent markers to be too narrow on portrait layout charts
- fatter markers look better (according to @andraskangyal)

Reply:
1:1 has doesn't make sense. Sqrt makes 1:1

Reply:
Or this means keepaspectratio should be true?

Reply:
It means that now the grid will be between 1:1 -- 16:9. We should change that to a symmetric range around 1:1 to achieve the 2 points I wrote in the first comment. E.g. 3:4 -- 4:3

Reply:
How can we decide that it should be portrait or landscape? There are no information about canvas aspect ratio here.

Reply:
If we want to make it 4:3 then 1:1- 4:3 is the only option here (3:4-4:3 doesn't make any sense, because no x-y order exists). The other option is the keepAspectRatio

Reply:
OK, so the 2 problems I mentioned in the comment are also a disadvantage of this solution which we did not consider during the decision. @andraskangyal please give this another thought.

Reply:
note: the current solution on portrait aspect ratio, the markers are portrait too (so taller than wider), so currently those are not "fatter" at all.

Reply:
Fatter then the proposed solution.

Reply:
but with 16:9 -> 1:1 and keepAspectRatio can solve this.

Reply:
But it introduces yet another problem, we won't fill up the available plot area space

Reply:
We discussed this again with @andraskangyal, considering the disadvantages of all solutions: 
- old version: less marker in the last row
- new version: makes the markers narrower, which is not desired in portrait layout
- new version + keep aspect ratio: makes empty space above/below or left/right side of the chart
we should keep the old version.

Reply:
Also asked @dovicsin to research possible visualizations to represent pure dimensional data. Maybe table chart not our best option for this job on the first place.